skbio/sequence/_feature.py

```diff
@@ -0,0 +1,52 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+from collections import Mapping
+import numpy as np
+
+
+class Feature(Mapping):
+    '''Store the metadata of a sequence feature.
```

Comment:
we should enforce the presence of some keys including strand, left (open in left end), right (open in right end), type (feature type) and give them default value if not set, as these info should be required for every feature.